paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/CustomerRepositoryKtx.kt

```diff
@@ -0,0 +1,73 @@
+package com.stripe.android.paymentsheet.repositories
+
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * Removes the provided saved payment method alongside any duplicate stored payment methods. This function should be
+ * deleted once an endpoint is stood up to handle detaching and removing duplicates.
+ *
+ * @param customerInfo authentication information that can perform detaching operations.
+ * @param paymentMethodId the id of the payment method to remove and to compare with for stored duplicates
+ *
+ * @return a list of removal results that identify the payment methods that were successfully removed and the payment
+ * methods that failed to be removed.
+ */
+internal suspend fun CustomerRepository.detachPaymentMethodAndDuplicates(
+    customerInfo: CustomerRepository.CustomerInfo,
+    paymentMethodId: String
+): List<PaymentMethodRemovalResult> {
+    val paymentMethods = getPaymentMethods(
+        customerInfo = customerInfo,
+        // We only support removing duplicate cards.
+        types = listOf(PaymentMethod.Type.Card),
+        silentlyFail = false,
+    ).getOrElse {
+        return listOf(
+            PaymentMethodRemovalResult(
+                paymentMethodId = paymentMethodId,
+                result = Result.failure(it)
+            )
+        )
+    }
+
+    val requestedPaymentMethodToRemove = paymentMethods.find { paymentMethod ->
+        paymentMethod.id == paymentMethodId
+    } ?: return listOf()
```

Comment:
should this represent a failure of some sort? My interpretation of this is that someone requested that a payment method be removed and we didn't do anything. 

Does this actually just indicate that the PM isn't saved to the customer on the backend and so we don't need to do anything? Or does it indicate that we can't remove it properly?

Reply:
In this scenario is would do nothing. The Stripe API returns a failure if you try to detach a payment method that does not exist so I'll return a similar error.

Reply:
great, thanks!